lib/tasks/destroyed_location.rake

```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+namespace :destroyed_location do
+  desc 'Creates Destroyed Location Type and Location'
+  task create: :environment do |_t|
+    location_type_name = 'Destroyed'
+    location_name = 'Destroyed'
+    location_barcode = 'lw-destroyed'
+    parent_name = 'Sanger'
+    parent_type = 'Site'
+
+    ActiveRecord::Base.transaction do
+      # Needs a parent to be able to scan labware; use the Sanger Location
+      parent_type = LocationType.find_or_create_by!(name: parent_type)
+      parent = UnorderedLocation.create_with(
+        location_type: parent_type
+      ).find_or_create_by(name: parent_name)
+
+      # Create the Destroyed Location Type
+      location_type = LocationType.find_or_create_by!(name: location_type_name)
+
+      begin
+        # Use the specified Destroyed Location barcode; skip the callback to generate a barcode
+        UnorderedLocation.skip_callback(:create, :after, :generate_barcode)
```

Comment:
This is cool! Learnt something new.

Reply:
This presents an issue with the callback, if given a barcode it still overrides this which doesn't sound right. Not sure if thats worth changing because im not sure of the consequences.

Reply:
I have removed skipping/restoring the callback. Now the generate_barcode method will not generate and overwrite the location barcode if it has a barcode already. I have added a test for that.